examples/nodejs-nestjs/src/arcjet/arcjet.guard.ts

```diff
@@ -0,0 +1,44 @@
+import { CanActivate, ExecutionContext, HttpException, HttpStatus, Inject, Injectable, Optional } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import arcjet from '@arcjet/node';
+import { ARCJET_DEFAULT_RULES } from '../config/arcjet.js';
+
+@Injectable()
+export class ArcjetGuard implements CanActivate {
+    private aj: any;
+
+    constructor(
+        private configService: ConfigService,
+        @Optional() @Inject('ARCJET_RULES') private rules: any[] = []
+    ) {
+        this.aj = arcjet({
+            key: this.configService.get<string>('ARCJET_KEY'),
+            rules: [...ARCJET_DEFAULT_RULES, ...this.rules],
+        });
+    }
+
+    async canActivate(
+        context: ExecutionContext,
+    ): Promise<boolean> {
+        const request = context.switchToHttp().getRequest();
+
+        // const aj = arcjet({
```

Comment:
I suggest explaining why this is commented out and when/why you might want to use it.